packages/opentelemetry-plugin-http/src/http.ts

```diff
@@ -380,8 +380,12 @@ export class HttpPlugin extends BasePlugin<Http> {
         extraOptions
       );
 
+      if (utils.isOpenTelemetryRequest(optionsParsed)) {
+        delete optionsParsed.headers![utils.OT_REQUEST_HEADER];
```

Comment:
Instead of asserting, can we do `if (optionsParsed.headers && utils.ioOpen...`

Reply:
We could but that's checked anyway by the `isOpenTelemetryRequest`, i tried to use typeguard without success. I will retry to avoid asserting

Reply:
Fixed it !